### PR TITLE
feat: upgraded Kochava(7.1.0) and Rudder(1.12.0) SDK

### PR DESCRIPTION
**Pull request by pallabmaiti**
# Description

Upgraded Kochava(7.1.0) and Rudder(1.12.0) SDK

### CODE REVIEW

File: Rudder-Kochava/Classes/RudderKochavaIntegration.m [hunk 1]
```diff
@@ -78,215 +64,163 @@ - (void)flush {
...
-        return eventProperties;
-    }
-    if (eventProperties[@"productId"])
-    {
-        event.contentIdString = (NSString*)eventProperties[@"productId"];
-        [eventProperties removeObjectForKey:@"productId"];
+    if (eventProperties[KeyProductId]) {
+        event.contentIdString = (NSString*)eventProperties[KeyProductId];
+        [eventProperties removeObjectForKey:KeyProductId];
     }
```
[the remark this answers is not in the document]
Comment (desusai7) [reply Rudder-Kochava/Classes/RudderKochavaIntegration.m:191]: Let's make the same changes on Android as well

File: Rudder-Kochava/Classes/RudderKochavaIntegration.m [hunk 2]
```diff
@@ -78,215 +64,163 @@ - (void)flush {
...
-            {
+    if (products) {
+        for(NSDictionary *product in products) {
+            if (product[type]) {
                 [productProperties addObject:product[type]];
             }
-            if ([type isEqual:@"product_id"] && product[@"productId"])
-            {
-                [productProperties addObject:product[@"productId"]];
-            }
```
Comment (desusai7) [inline Rudder-Kochava/Classes/RudderKochavaIntegration.m:281]: Let's have same behaviour on both android and ios

File: Rudder-Kochava/Classes/RudderKochavaIntegration.m [hunk 3]
```diff
@@ -18,52 +18,38 @@ @implementation RudderKochavaIntegration
...
-            if (self.appGUID!=nil)
-            {
-                if ([[config objectForKey:@"appTrackingTransparency"] boolValue])
-                {
-                    KVATracker.shared.appTrackingTransparency.enabledBool= YES;
-                }
-                if ([[config objectForKey:@"skAdNetwork"] boolValue])
-                {
-                    [KVAAdNetworkProduct.shared register];
-                }
```
[the remark this answers is not in the document]
Comment (pallabmaiti) [reply Rudder-Kochava/Classes/RudderKochavaIntegration.m:41]: Text updated on integration config

File: Rudder-Kochava/Classes/RudderKochavaIntegration.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (1abhishekpandey) [reply Rudder-Kochava/Classes/RudderKochavaIntegration.m:190]: I've reverted the changes and now Kochava iOS supports both `product_id` and `productId`. In Kochava-v2, we will make the required changes.

File: Rudder-Kochava/Classes/RudderKochavaIntegration.m [hunk 4]
```diff
@@ -78,215 +64,163 @@ - (void)flush {
...
+                            event.ratingValueDoubleNumber = (NSNumber*)eventProperties[KeyRating];
+                            [eventProperties removeObjectForKey:KeyRating];
                         }
                     }
-                    if ([eventName isEqual:@"products searched"])
-                    {
-                        if (eventProperties[@"query"])
-                        {
-                            event.uriString = (NSString*) eventProperties[@"query"];
-                            [eventProperties removeObjectForKey:@"query"];
```
[the remark this answers is not in the document]
Comment (1abhishekpandey) [reply Rudder-Kochava/Classes/RudderKochavaIntegration.m:135]: We are creating a copy of the message object over [here](https://github.com/rudderlabs/rudder-integration-kochava-ios/blob/5456b6d8d86d3ef1ac50249461932ea0eed6ac22/Rudder-Kochava/Classes/RudderKochavaIntegration.m#L77).

File: Rudder-Kochava/Classes/RudderKochavaIntegration.m (hunk 2 above)
Comment (1abhishekpandey) [reply Rudder-Kochava/Classes/RudderKochavaIntegration.m:281]: I've refactored this logic.

**Pull request closed without merging**